### PR TITLE
Plans: Show trial label instead of plan name in sidebar when current plan is on free trial

### PR DESCRIPTION
**Pull request by stephanethomas**
This pull request fixes https://github.com/Automattic/wp-calypso/issues/1659 by displaying `Trial` instead of the plan name next to the `Plan` menu item in the sidebar when the current plan is on trial:

screenshot
 
#### Testing instructions

1. Run `git checkout fix/sidebar-plan-label` and start your server
2. Get a new test account and blog
3. Enable your proxy
5. Select a plan by clicking the `Start Free Trial` button on the [`Plans` page](http://calypso.localhost:3000/plans)
6. Click the `Start 14 Day Free Trial` button on the `Secure Payment` page
7. Check that the purchase was successful
8. Navigate to any page of the `My Sites` section
9. Check that `Trial` is displayed in the sidebar
10. Upgrade your plan from the `Plans` page
11. Check that the plan name is now displayed in the sidebar

#### Reviews

- [x] Code
- [x] Product

### CODE REVIEW

File: client/my-sites/sidebar/sidebar.jsx [hunk 1]
```diff
@@ -308,14 +308,20 @@ module.exports = React.createClass( {
...
-			planName = ( this.isSingle() ) ? site.plan.product_name_short : '';
+		const planLink = '/plans' + this.siteSuffix();
 
 		let linkClass = 'upgrades-nudge';
+
 		if ( productsValues.isPlan( site.plan ) ) {
 			linkClass += ' is-paid-plan';
 		}
 
+		let planName = site.plan.product_name_short;
```
Comment (scruffian) [inline client/my-sites/sidebar/sidebar.jsx]: Did you ever work out why there was a call to `this.isSingle()` previously? It seems to not be needed, but I wonder if there's an explanation...

**Comment by scruffian** [comment]
Code :+1:

File: client/my-sites/sidebar/sidebar.jsx (hunk 1 above)
Comment (stephanethomas) [inline client/my-sites/sidebar/sidebar.jsx]: I guess it's just the result of this [copy/paste](https://github.com/Automattic/wp-calypso/commit/e5b0c0c93c7cd35f44707ab84c7901f99ccd8caa#diff-3e29fa10a6f1fb8927b10336ff9ff66bL281).

**Comment by spncrb** [comment]
Tested for business and Premium - seems to work fine - including user facing cancellation :+1:

**Pull request merged**